### DISCUSSION TITLE
How to ignore/blocklist a regex or subtree of folders entirely in config file?

### DISCUSSION THREAD

**Discussion by real-or-random** [emacs-lsp/lsp-mode#4387] [Q&A]
How can I configure a regexp (or a subtree in my file system) in which `lsp-mode` are ignored?

When I open a file in a workspace for the first time, I get a menu and I can press `d` or even `D` to select the folder. That's good so far, but where is this setting stored? I'd prefer to include a simple two-liner in my emacs config file to enhance the maintainability of my config. 

The menu refers to  `lsp-session-folders-blocklist`, but it's a function and not a variable. Well, okay, it's a getter, the docstring is:

> Access slot "folders-blocklist" of lsp-session struct CL-X.

and then I see this:

```
(cl-defstruct lsp-session
[...]
  ;; contains the folders that must not be imported in the current workspace.
  folders-blocklist
[...] )


This sounds like something specific to a workspace already, and not something that works globally?

**Comment by yyoncho** [comment]
Something like that should work:

```
(with-eval-after-load 'lsp-mode
  (cl-pushnew "/foo" (lsp-session-folders-blocklist (lsp-session))
              :test 'string=))
```

you typically have one lsp session per emacs session.

**Reply by real-or-random** [reply]
Thanks a lot! That doesn't do the trick yet.

I suspect that it runs too early. I get this when I open the first file where lsp is supposed to run (whether it's in the blocklisted folder or not), and `lsp-session-folders-blocklist` wasn't updated:
```
Error (python-mode-local-vars-hook): Error running hook "lsp!" because: (void-function (setf lsp-session-folders-blocklist))
```

This may be a doom emacs thing, `lsp!` is a doom function that initializes lsp support by dispatching to lsp-mode or eglot.  Any further ideas?
